src/components/ImportColumn.tsx

```diff
@@ -0,0 +1,182 @@
+import {setColumnName} from '@libs/actions/ImportSpreadsheet';
+import CONST from '@src/CONST';
+import ButtonWithDropdownMenu from './ButtonWithDropdownMenu';
+import Text from './Text';
+
+const findColumnName = (header: string) => {
+    let attribute = '';
+    const formattedHeader = Str.removeSpaces(String(header).toLowerCase().trim());
+    switch (formattedHeader) {
+        case 'email':
+        case 'emailaddress':
+        case 'emailaddresses':
+        case 'e-mail':
+        case 'e-mailaddress':
+        case 'e-mailaddresses':
+            attribute = CONST.CSV_IMPORT_COLUMNS.EMAIL;
+            break;
+
+        case 'category':
+        case 'categories':
+            attribute = CONST.CSV_IMPORT_COLUMNS.EMAIL;
+            break;
+
+        case 'glcode':
+        case 'gl':
+            attribute = CONST.CSV_IMPORT_COLUMNS.GL_CODE;
+            break;
+
+        case 'tag':
+        case 'tags':
+        case 'project':
+        case 'projectcode':
+        case 'customer':
+        case 'name':
+            attribute = 'name';
+            break;
+
+        case 'submitto':
+        case 'submitsto':
+            attribute = CONST.CSV_IMPORT_COLUMNS.SUBMIT_TO;
+            break;
+
+        case 'approveto':
+        case 'approvesto':
+            attribute = CONST.CSV_IMPORT_COLUMNS.APPROVE_TO;
+            break;
+
+        case 'payroll':
+        case 'payrollid':
+        case 'payrolls':
+        case 'payrol':
+        case 'customfield2':
+            attribute = CONST.CSV_IMPORT_COLUMNS.CUSTOM_FIELD_2;
+            break;
+
+        case 'userid':
+        case 'customfield1':
+            attribute = CONST.CSV_IMPORT_COLUMNS.CUSTOM_FIELD_1;
+            break;
+
+        case 'role':
+            attribute = CONST.CSV_IMPORT_COLUMNS.ROLE;
+            break;
+
+        case 'total':
+        case 'threshold':
+        case 'reporttotal':
+        case 'reporttotalthreshold':
+        case 'approvallimit':
+            attribute = CONST.CSV_IMPORT_COLUMNS.REPORT_THRESHHOLD;
+            break;
+
+        case 'alternate':
+        case 'alternateapprove':
+        case 'alternateapproveto':
+        case 'overlimitforwardsto':
+            attribute = CONST.CSV_IMPORT_COLUMNS.APPROVE_TO_ALTERNATE;
+
+            break;
+
+        case 'destination':
+            attribute = CONST.CSV_IMPORT_COLUMNS.NAME;
+            break;
+
+        case 'subrate':
+            attribute = CONST.CSV_IMPORT_COLUMNS.SUBRATE;
+            break;
+
+        case 'amount':
+            attribute = CONST.CSV_IMPORT_COLUMNS.AMOUNT;
+            break;
+
+        case 'currency':
+            attribute = CONST.CSV_IMPORT_COLUMNS.CURRENCY;
+            break;
+
+        case 'rateid':
+            attribute = CONST.CSV_IMPORT_COLUMNS.RATE_ID;
+            break;
+
+        case 'enabled':
+        case 'enable':
+            attribute = CONST.CSV_IMPORT_COLUMNS.ENABLED;
+            break;
+
+        default:
+            break;
```

Comment:
This means we can return an empty string, should we throw instead?

Reply:
No big difference I think